code/game/objects/items/weapons/storage/bags.dm

```diff
@@ -479,15 +479,17 @@
 		var/dropped_something = FALSE
 
 		for(var/obj/item/I in contents)
-			I.loc = dropspot
+			var/obj/item/newItem = new I.type(dropspot)
+			// Set the properties of the new item here, e.g., stack count, hover highlight, tooltip
 			contents.Remove(I)
+			qdel(I) // Delete the old item
```

Comment:
Deleting an item and replacing it with the same type is a terrible solution for this. This should use something like `remove_from_storage` to properly remove the maptext.